### PR TITLE
Remove old xfails from kes tests

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

**Review by mkoura — CHANGES_REQUESTED** [review]
We still want the `check_kes_period_info_result` to collect errors, not to abort tests. That would happen when calling `.finish_test()`.

I think we could leave the code as it was, and just add "blocking" checks into `get_xfails` for each potential xfail error, like

```python
if "issue_4114" in error and blockers.GH(issue=4114).is_blocked():
    ...
```

**Comment by saratomaz** [comment]
I renamed the function to avoid confusion with pytest xfail, since the known not blocked issues will fail.

File: cardano_node_tests/tests/kes.py [hunk 1]
```diff
@@ -182,20 +185,20 @@ def check_kes_period_info_result(  # noqa: C901
     return errors
 
 
-def get_xfails(errors: List[str]) -> Set[str]:
-    """Get xfail error strings."""
-    xfails = set()
+def get_known_fails(errors: List[str]) -> Set[str]:
```
Comment (mkoura) [inline cardano_node_tests/tests/kes.py:188]: What you are getting here are really XFAILs :-) So there's no need to rename this.

**Review by mkoura — CHANGES_REQUESTED** [review]

File: cardano_node_tests/tests/test_kes.py [hunk 2]
```diff
@@ -804,9 +804,9 @@ def test_update_valid_opcert(
 
         err_joined = "\n".join(e for e in kes_period_info_errors_list if e)
         if err_joined:
-            xfails = kes.get_xfails(errors=kes_period_info_errors_list)
-            if xfails:
-                pytest.xfail(" ".join(xfails))
+            known_fails = kes.get_known_fails(errors=kes_period_info_errors_list)
+            if known_fails:
+                pytest.fail(" ".join(known_fails))
```
Comment (mkoura) [inline cardano_node_tests/tests/test_kes.py:809]: This should be `pytest.xfail`

File: cardano_node_tests/tests/test_staking_no_rewards.py [hunk 3]
```diff
@@ -1002,8 +1002,8 @@ def test_deregister_reward_addr_retire_pool(
 
         err_joined = "\n".join(e for e in kes_period_info_errors_list if e)
         if err_joined:
-            xfails = kes.get_xfails(errors=kes_period_info_errors_list)
-            if xfails:
-                pytest.xfail(" ".join(xfails))
+            known_fails = kes.get_known_fails(errors=kes_period_info_errors_list)
+            if known_fails:
+                pytest.fail(" ".join(known_fails))
```
Comment (mkoura) [inline cardano_node_tests/tests/test_staking_no_rewards.py:1007]: This should be `pytest.xfail`

**Review by mkoura — APPROVED** [review]

**Pull request merged**